test/hotspot/jtreg/runtime/Thread/StopAtExit.java

```diff
@@ -93,7 +93,15 @@ public static void main(String[] args) {
             // the thread is terminated.
             ThreadGroup myTG = new ThreadGroup("myTG-" + count);
             myTG.setDaemon(true);
-            Throwable myException = new ThreadDeath();
+            Throwable myException;
+            if ((count % 1) == 1) {
+              // Throw RuntimeException before ThreadDeath since a
+              // ThreadDeath can also be queued up when there's already
+              // a non-ThreadDeath async execution queued up.
+              myException = new RuntimeException();
+            } else {
+              myException = new ThreadDeath();
+            }
```

Comment:
I think this should go in the inner loop if we want to throw different exceptions for the same target.

Reply:
Thanks for catching this also!